hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1159,6 +1159,8 @@ public void deleteChangedReaderObserver(ChangedReadersObserver o) {
       }
     }
     replaceStoreFiles(filesToCompact, sfs, true);
+    storeEngine.resetCompactionWriter();
```

Comment:
Leave a big fat comment here as this is critical to the correctness of the BrokenStoreFileCleanerChore.